src/error.rs

```diff
@@ -127,6 +244,8 @@ where
         Self {
             error: anyhow::Error::new(error),
             status: StatusCode::InternalServerError,
+            headers: None,
```

Comment:
This part confuses me. If you can losslessly turn an Error into a Response, why do Errors need to be "response-like" by having headers and body? It seems like it would be preferable just to transform them into Responses instead of duplicating header and body logic on Error

Reply:
I'm not sure what you mean - in order to "losslessly" turn an `Error` into a `Response`, you'd need this information. The `Response.error` portion of this PR is secondary, the main idea is to be able to propagate an error as a distinct `Error` with enough information to make a `Response` back to any potential client once any middleware is finished with it.